### DISCUSSION TITLE
How to set StrictHostKeyChecking "no"?

### DISCUSSION THREAD

**Discussion by garak** [deployphp/deployer#2270] [Help needed]
Using version 7-beta3, with this config:

```
host('foo')
    ->setHostname('myIpAddress')
    ->set('deploy_path', '~/projects/{{application}}')
;
```

I'm getting problem about "authenticity of host can't be established", with question "you sure you want to continue connecting (yes/no)?"
In a context of CI, where I can't reply to such question, how can I set `StrictHostKeyChecking=no`?

I tried with this:
```
host('foo')
    ->setHostname('myIpAddress')
    ->set('deploy_path', '~/projects/{{application}}')
    ->setSshArguments(['StrictHostKeyChecking' => 'no'])
;
```

but it doesn't work, I'm getting error `ssh: Could not resolve hostname no: No address associated with hostname`

**Comment by antonmedv** [comment]
In Deployer v7 ssh options were refactored. You can do it like this:
```php
host('deployer.org')
    ->setSshArguments(['-o StrictHostKeyChecking=no']);
```
There is always `-vvv` option to debug what Deployer is doing in the background.

PS Docs updates will be soon, with a better explanation of why this refactoring was introduced.

**Comment by garak** [comment]
I'm still in trouble...

```
host('foo')
    ->setHostname('myIpAddress')
    ->set('deploy_path', '~/projects/{{application}}')
    ->set('user', 'myuser')
    ->setSshArguments(['-o StrictHostKeyChecking=no -o BatchMode=yes'])
;
```

but I see (using `-vvv`) the following command is issued during connect:

> ssh -N  -o StrictHostKeyChecking=no -o BatchMode=yes -A -o ControlMaster=auto -o ControlPersist=60 -o ControlPath=/dev/shm/%C myIpAddress

resulting in an access denied error, because the user is not specified

**Reply by antonmedv** [reply]
Replace user with remote_user

**Comment by martijngastkemper** [comment]
Instead of disabling the host key check you can add the host key before executing deployer by running: `ssh-keyscan myIpAddress >> ~/.ssh/known_hosts`.
